test/framework/finalizers_helpers.go

```diff
@@ -125,7 +127,11 @@ func getObjectsWithFinalizers(ctx context.Context, proxy ClusterProxy, namespace
 
 		if len(setFinalizers) > 0 {
```

Comment:
(I'm aware this was the same before)

This only validates finalizers if there are actually finalizers on the object, which means missing finalizers are accepted

Do I understand it correctly that basically we assume the finalizers are correct the first time we call getObjectsWithFinalizers and then in assertFinalizersExist we would detect if a finalizer is missing compared to the first time we retrieved them?

Reply:
That's correct.
I have noticed this as well and now it should be fixed (we also check for missing finalizers + we check for additional finalizers that can pop up after we remove them)

Reply:
Thx, nice one!